SwiftCompilerSources/Sources/Optimizer/Utilities/Cloner.swift

```diff
@@ -0,0 +1,186 @@
+//===--- Cloner.swift ------------------------------------------------------==//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import OptimizerBridging
+import SIL
+import SILBridging
+
+/// Clones the initializer value of a GlobalVariable.
+///
+/// Used to transitively clone "constant" instructions, including their operands,
+/// from or to the static initializer value of a GlobalVariable.
+///
+struct Cloner<Context: MutatingContext> {
```

Comment:
One more thing: can you move the Cloner (Cloner.swift, the cloner bridging code) into the SIL module?